src/ui/pages/SidePage/components/IncomingRequest/components/SignRequest.tsx

```diff
@@ -21,22 +22,29 @@ const SignRequest = ({
   handleAccept,
   handleCancel,
 }: RequestProps) => {
-  const signRequest = requestData.signTransaction;
   const [isSigningObject, setIsSigningObject] = useState(false);
-  const logo = requestData.logo ? requestData.logo : CardanoLogo;
-
   const signDetails = useMemo(() => {
-    if (!signRequest) return {};
+    if (
+      requestData.type !== IncomingRequestType.PEER_CONNECT_SIGN ||
+      !requestData.signTransaction
+    )
+      return {};
 
     let signContent;
     try {
-      signContent = JSON.parse(signRequest.payload.payload);
+      signContent = JSON.parse(requestData.signTransaction.payload.payload);
       setIsSigningObject(true);
     } catch (error) {
-      signContent = signRequest.payload.payload;
+      signContent = requestData.signTransaction.payload.payload;
     }
     return signContent;
-  }, [requestData.signTransaction]);
+  }, [requestData.type]);
+
+  if (requestData.type !== IncomingRequestType.PEER_CONNECT_SIGN) {
+    return null;
+  }
+  const signRequest = requestData.signTransaction;
+  const logo = CardanoLogo;
```

Comment:
This should come from the peer connection. And actually I think the fallback should be the generic user icon not the Cardano logo - let me check with Robert on this

Reply:
Confirmed: logo comes from the peer connection. If the logo is undefined, fallback to the generic user icon (this is used when we don't have the logo in the list of connections already)